bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/FindReplaceOverlay.java

```diff
@@ -167,15 +175,48 @@ private void setTextEditorActionsActivated(boolean state) {
 			if (!(targetPart instanceof AbstractTextEditor)) {
 				return;
 			}
+			if (targetPart.getSite() instanceof MultiPageEditorSite multiEditorSite) {
+				if (!state && globalActionHandlerDeaction == null) {
+					globalActionHandlerDeaction = new DeactivateGlobalActionHandlers(multiEditorSite.getActionBars());
+				} else if (state && globalActionHandlerDeaction != null) {
+					globalActionHandlerDeaction.reactivate();
+					globalActionHandlerDeaction = null;
+				}
+			}
 			try {
 				Method method = AbstractTextEditor.class.getDeclaredMethod("setActionActivation", boolean.class); //$NON-NLS-1$
 				method.setAccessible(true);
 				method.invoke(targetPart, Boolean.valueOf(state));
-			} catch (IllegalArgumentException | InvocationTargetException | IllegalAccessException | SecurityException | NoSuchMethodException ex) {
+			} catch (IllegalArgumentException | InvocationTargetException | IllegalAccessException | SecurityException
+					| NoSuchMethodException ex) {
 				TextEditorPlugin.getDefault().getLog()
 						.log(Status.error("cannot (de-)activate actions for text editor", ex)); //$NON-NLS-1$
 			}
 		}
+
+		static final class DeactivateGlobalActionHandlers {
+			private final static String[] ACTIONS = { ITextEditorActionConstants.CUT, ITextEditorActionConstants.COPY,
+					ITextEditorActionConstants.PASTE, ITextEditorActionConstants.DELETE,
+					ITextEditorActionConstants.SELECT_ALL, ITextEditorActionConstants.FIND };
```

Comment:
Maybe you make this more modern?
```suggestion
			private final static List<String> ACTIONS = List.of(ITextEditorActionConstants.CUT,
					ITextEditorActionConstants.COPY, ITextEditorActionConstants.PASTE,
					ITextEditorActionConstants.DELETE, ITextEditorActionConstants.SELECT_ALL,
					ITextEditorActionConstants.FIND);
```

Reply:
That's a good idea. I had just copied this over and adapted it, but of couse makes sense to improve it as well. Done with 5c7a21c